packages/loader/container-loader/src/container.ts

```diff
@@ -1777,7 +1779,10 @@ export class Container
 	private get client(): IClient {
 		const client: IClient =
 			this.options?.client !== undefined
-				? (this.options.client as IClient)
+				? // this.options.client is shared by ALL containers spawned by the same Loader (including the summarizer)
+				  // We're doing a cheap clone here to ensure it doesn't get tainted by the merge() that happens below, since
+				  // the override should only apply to the individual container it is given to.
+				  (JSON.parse(JSON.stringify(this.options.client)) as IClient)
```

Comment:
this will clone on every access. if we go this route, we should cache the value.

Reply:
Longer-term we should remove this as an option entirely, so if your concern is performance I think I'm inclined to punt and then go make it moot by removing the option.

I think this is the lower-risk approach without further research since the participating objects (this.options.client and this.clientDetailsOverride) have historically both remained mutable for the lifetime of the container.  Cloning here ensures any weird patterns that depended on that characteristic continue to function the same.  Not saying they should mutate them, but given that we're trying to patch this quickly I prefer to take the minimal deviation from historical behavior.

Reply:
FWIW - this is only called in a few places, mostly during startup a single time.  Only recurring callsites are
1. If attach() needs to retry, it calls once per retry
2. When connecting, calls once per connectToDeltaStream()

I'm also realizing now that this mutation was why client.details.environment always seemed so spammy.

Reply:
> the participating objects (this.options.client and this.clientDetailsOverride) have historically both remained mutable for the lifetime of the container. Cloning here ensures any weird patterns that depended on that characteristic continue to function the same

I'm confused.  As you have it, the client returned by the getter will be a different object every time, which would
thwart anyone counting on container-global mutability, right?  So wouldn't it be better by that argument to cache one clone of `this.options.client` and reuse that, just as previously it was reusing a single object for the whole loader?  Or am I mixed up?

Reply:
Ok we chatted about it - What I was confused on is _who_ is doing the mutating.  I was thinking you meant consumers of the `private get client()` in `Container`, but you're talking about external actors who are mutating the one on the loader.  Makes sense